internal/test/builder/builders.go

```diff
@@ -394,6 +394,7 @@ func (m *MachineDeploymentClassBuilder) Build() *clusterv1.MachineDeploymentClas
 }
 
 // InfrastructureMachineTemplateBuilder holds the variables and objects needed to build an InfrastructureMachineTemplate.
+// +kubebuilder:object:generate=false
```

Comment:
Just for my understanding, why are we excluding some builder from generating deep copy?

Reply:
The generation didn't work for map[string]interface{} - I'm not sure if there's a way to manage this automatically. This was the simplest solution as we don't strictly need DeepCopy on these types yet, but I'd be happy to fix it. WDYT?

Reply:
Thanks for the explanation.
If we will need DeepCopy for those types we can either implement it manually or consider using something like apiextensionsv1.JSON (there are also other options), but I ok in tacking thin in a follow up iteration

Reply:
Using something like apiextensionsv1.JSON sounds reasonable to me, too. Let's open an issue?